Test passes in GH Actions but fails locally

Locally, `tests/pytest/streamlit_app/test_utils.py::test__discover_apps` fails
```
============================= test session starts ==============================
platform linux -- Python 3.12.10, pytest-8.3.5, pluggy-1.5.0
django: version: 5.1.6, settings: pems.settings (from ini)
rootdir: /caltrans/app
configfile: pyproject.toml
plugins: django-4.11.1, mock-3.14.0, socket-0.7.0
collected 15 items

tests/pytest/pems/district/test_models.py .                              [  6%]
tests/pytest/pems/core/test_middleware_healthcheck.py .                  [ 13%]
tests/pytest/streamlit_app/test_main.py ....                             [ 40%]
tests/pytest/streamlit_app/test_utils.py ..F......                       [100%]
=================================== FAILURES ===================================
_____________________________ test__discover_apps ______________________________

app_paths = [PosixPath('/caltrans/app/tests/pytest/streamlit_app/test_apps/app_one.py'), PosixPath('/caltrans/app/tests/pytest/streamlit_app/test_apps/app_two.py')]

    def test__discover_apps(app_paths):
        discovered = utils._discover_apps()
    
>       assert discovered == app_paths
E       AssertionError: assert [PosixPath('/.../app_one.py')] == [PosixPath('/.../app_two.py')]
E         
E         At index 0 diff: PosixPath('/caltrans/app/tests/pytest/streamlit_app/test_apps/app_two.py') != PosixPath('/caltrans/app/tests/pytest/streamlit_app/test_apps/app_one.py')
E         Use -v to get more diff

tests/pytest/streamlit_app/test_utils.py:47: AssertionError
```

but it passes in [GH Actions](https://github.com/compilerla/pems/actions/runs/14494387829/job/40658238693)
```
============================= test session starts ==============================
platform linux -- Python 3.12.10, pytest-8.3.5, pluggy-1.5.0
django: version: 5.1.6, settings: pems.settings (from ini)
rootdir: /home/runner/work/pems/pems
configfile: pyproject.toml
plugins: socket-0.7.0, django-4.11.1, mock-3.14.0
collected 15 items

tests/pytest/pems/district/test_models.py .                              [  6%]
tests/pytest/pems/core/test_middleware_healthcheck.py .                  [ 13%]
tests/pytest/streamlit_app/test_main.py ....                             [ 40%]
tests/pytest/streamlit_app/test_utils.py .........                       [100%]

============================== 15 passed in 0.94s ==============================
Wrote HTML report to pems/static/coverage/index.html
```

We could change
https://github.com/compilerla/pems/blob/main/tests/pytest/streamlit_app/test_utils.py#L47
to 
```python
assert set(discovered) == set(app_paths)
```
so that the order of the items of the list doesn't matter, but it's still strange why we are seeing this difference.

--- a/tests/pytest/streamlit_app/test_utils.py
+++ b/tests/pytest/streamlit_app/test_utils.py
@@ -44,7 +44,7 @@ def test_default_app_page():
 def test__discover_apps(app_paths):
     discovered = utils._discover_apps()
 
-    assert discovered == app_paths
+    assert set(discovered) == set(app_paths)
 
 
 @pytest.mark.parametrize(

